api/app/src/main/kotlin/packit/service/RolePermissionService.kt

```diff
@@ -2,83 +2,182 @@ package packit.service
+    internal fun removeRolePermissionsFromRole(role: Role, removeRolePermissions: List<UpdateRolePermission>)
     {
         val rolePermissionsToRemove = getRolePermissionsToUpdate(role, removeRolePermissions)
-        var rolePermissionsToRemoveIds = mutableListOf<Int>()
 
         for (rolePermission in rolePermissionsToRemove)
         {
             val matchedRolePermission = role.rolePermissions.find { rolePermission == it }
                 ?: throw PackitException("rolePermissionDoesNotExist", HttpStatus.BAD_REQUEST)
 
-            rolePermissionsToRemoveIds.add(matchedRolePermission.id!!)
             role.rolePermissions.remove(matchedRolePermission)
         }
-
-        rolePermissionRepository.deleteAllByIdIn(rolePermissionsToRemoveIds)
-        return role
     }
 
-    override fun getRolePermissionsToAdd(
+    internal fun addPermissionsToRole(
         role: Role,
-        addRolePermissions: List<UpdateRolePermission>
-    ): List<RolePermission>
+        addRolePermissions: List<UpdateRolePermission>,
+    )
     {
         val rolePermissionsToAdd = getRolePermissionsToUpdate(role, addRolePermissions)
         if (rolePermissionsToAdd.any { role.rolePermissions.contains(it) })
         {
             throw PackitException("rolePermissionAlreadyExists", HttpStatus.BAD_REQUEST)
         }
+        role.rolePermissions.addAll(rolePermissionsToAdd)
+    }
 
-        return rolePermissionsToAdd
+    internal fun addPermissionToRoles(
+        roles: List<Role>,
+        permission: Permission,
+        packet: Packet? = null,
+        packetGroup: PacketGroup? = null,
+        tag: Tag? = null,
+    )
+    {
+        for (role in roles)
+        {
+            val rolePermission = RolePermission(
+                role = role,
+                permission = permission,
+                packet = packet,
+                packetGroup = packetGroup,
+                tag = tag,
+            )
+            if (role.rolePermissions.contains(rolePermission))
+            {
+                throw PackitException("rolePermissionAlreadyExists", HttpStatus.BAD_REQUEST)
+            }
+            role.rolePermissions.add(rolePermission)
+        }
+    }
+
+    internal fun removePermissionFromRoles(
+        roles: List<Role>,
+        permission: Permission,
+        packet: Packet? = null,
+        packetGroup: PacketGroup? = null,
+        tag: Tag? = null,
+    )
+    {
+        for (role in roles)
+        {
+            val rolePermission = RolePermission(
+                role = role,
+                permission = permission,
+                packet = packet,
+                packetGroup = packetGroup,
+                tag = tag,
+            )
+            val matchedRolePermission = role.rolePermissions.find { it == rolePermission }
+                ?: throw PackitException("rolePermissionDoesNotExist", HttpStatus.BAD_REQUEST)
+
+            role.rolePermissions.remove(matchedRolePermission)
```

Comment:
```suggestion
            if (!role.rolePermissions.remove(rolePermission)) {
                throw PackitException("rolePermissionDoesNotExist", HttpStatus.BAD_REQUEST)
            }
```

List's remove gives you back a boolean: https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/-mutable-list/remove.html

Same applies to the `removeRolePermissionsFromRole` method